modules/code-editor/src/backend/editor.ts

```diff
@@ -197,14 +197,20 @@ export default class Editor {
       return this._getGhost(file).upsertFile('/actions', location, content, true, true)
     }
 
-    const ghost = this.bp.ghost.forGlobal()
```

Comment:
If everything is validated correctly, you could just use this._getGhost(file) to get the correctly-scoped ghost for the operation (like type === 'action)

Reply:
+1